tests/pytests/unit/test_master.py

```diff
@@ -160,3 +202,749 @@ def test_when_syndic_return_processes_load_then_correct_values_should_be_returne
+            "name": "AuthorizationError",
+            "message": "Authorization error occurred.",
+        }
+    }
+    with patch(
+        "salt.acl.PublisherACL.user_is_blacklisted", MagicMock(return_value=False)
+    ), patch("salt.acl.PublisherACL.cmd_is_blacklisted", MagicMock(return_value=True)):
+        assert mock_ret == await clear_funcs.publish({"user": "foo", "fun": "test.arg"})
+
+
+@pytest.mark.slow_test
+async def test_publish_token_not_authenticated(clear_funcs):
+    """
+    Asserts that an AuthenticationError is returned when the token can't authenticate.
+    """
+    mock_ret = {
+        "error": {
+            "name": "AuthenticationError",
+            "message": "Authentication error occurred.",
+        }
+    }
+    load = {
+        "user": "foo",
+        "fun": "test.arg",
+        "tgt": "test_minion",
+        "kwargs": {"token": "asdfasdfasdfasdf"},
+    }
+    with patch(
+        "salt.acl.PublisherACL.user_is_blacklisted", MagicMock(return_value=False)
+    ), patch("salt.acl.PublisherACL.cmd_is_blacklisted", MagicMock(return_value=False)):
+        assert mock_ret == await clear_funcs.publish(load)
+
+
+@pytest.mark.slow_test
+async def test_publish_token_authorization_error(clear_funcs):
+    """
+    Asserts that an AuthorizationError is returned when the token authenticates, but is not
+    authorized.
+    """
+    token = "asdfasdfasdfasdf"
+    load = {
+        "user": "foo",
+        "fun": "test.arg",
+        "tgt": "test_minion",
+        "arg": "bar",
+        "kwargs": {"token": token},
+    }
+    mock_token = {"token": token, "eauth": "foo", "name": "test"}
+    mock_ret = {
+        "error": {
+            "name": "AuthorizationError",
+            "message": "Authorization error occurred.",
+        }
+    }
+
+    with patch(
+        "salt.acl.PublisherACL.user_is_blacklisted", MagicMock(return_value=False)
+    ), patch(
+        "salt.acl.PublisherACL.cmd_is_blacklisted", MagicMock(return_value=False)
+    ), patch(
+        "salt.auth.LoadAuth.authenticate_token", MagicMock(return_value=mock_token)
+    ), patch(
+        "salt.auth.LoadAuth.get_auth_list", MagicMock(return_value=[])
+    ):
+        assert mock_ret == await clear_funcs.publish(load)
+
+
+@pytest.mark.slow_test
+async def test_publish_eauth_not_authenticated(clear_funcs):
+    """
+    Asserts that an AuthenticationError is returned when the user can't authenticate.
+    """
+    load = {
+        "user": "test",
+        "fun": "test.arg",
+        "tgt": "test_minion",
+        "kwargs": {"eauth": "foo"},
+    }
+    mock_ret = {
+        "error": {
+            "name": "AuthenticationError",
+            "message": "Authentication error occurred.",
+        }
+    }
+    with patch(
+        "salt.acl.PublisherACL.user_is_blacklisted", MagicMock(return_value=False)
+    ), patch("salt.acl.PublisherACL.cmd_is_blacklisted", MagicMock(return_value=False)):
+        assert mock_ret == await clear_funcs.publish(load)
```

Comment:
```suggestion
        ret == await clear_funcs.publish(load) 
        assert ret == mock_ret
```
We should assert what we get against what we expect, it makes more sense when the test fails and pytest shows the diff.
All other tests seem to also have this reversed assertion.

Reply:
good point, fixed.